File: Ticket_2_Unit_Test_Examples/NotamUtils.java
import java.util.*;

public class NotamUtils {

    public static class Notam {
        public String id;
        public String type;
        public String location;
        public String urgency;

        public Notam(String id, String type, String location, String urgency) {
            this.id = id;
            this.type = type;
            this.location = location;
            this.urgency = urgency;
        }
    }

    public static List<Notam> fetchNotams() {
        List<Notam> notams = new ArrayList<>();
        notams.add(new Notam("A1", "Runway Closure", "KOKC", "normal"));
        notams.add(new Notam("B2", "Airspace Restriction", "KLAX", "high"));
        return notams;
    }

    public static Notam processNotam(Map<String, String> rawNotam) {
        String id = rawNotam.get("id");
        String type = rawNotam.get("type");
        String location = rawNotam.get("location");
        String urgency = rawNotam.getOrDefault("urgency", "normal");

        return new Notam(id, type, location, urgency);
    }
}

File: Ticket_2_Unit_Test_Examples/NotamUtils.py
from __future__ import annotations

from dataclasses import dataclass
from typing import Dict, List

@dataclass
class Notam:
    id: str
    type: str
    location: str
    urgency: str

def fetch_notams() -> List[Notam]:
    notams: List[Notam] = []
    notams.append(Notam("A1", "Runway Closure", "KOKC", "normal"))
    notams.append(Notam("B2", "Airspace Restriction", "KLAX", "high"))
    return notams

def process_notam(raw_notam: Dict[str, str]) -> Notam:
    notam_id = raw_notam.get("id")
    notam_type = raw_notam.get("type")
    location = raw_notam.get("location")
    urgency = raw_notam.get("urgency", "normal")

    return Notam(notam_id, notam_type, location, urgency)

